src/orion/benchmark/benchmark_client.py

```diff
@@ -0,0 +1,145 @@
+
+
+def get_or_create_benchmark(
+    name, algorithms=None, targets=None, storage=None, debug=False
+):
+    """
+    Create or get a benchmark object.
+
+    Parameters
+    ----------
+    name: str
+        Name of the benchmark
+    algorithms: list, optional
+        Algorithms used for benchmark, each algorithm can be a string or dict.
+    targets: list, optional
+        Targets for the benchmark, each target will be a dict with two keys.
+
+        assess: list
+            Assessment objects
+        task: list
+            Task objects
+    storage: dict, optional
+        Configuration of the storage backend.
+    debug: bool, optional
+        If using in debug mode, the storage config is overrided with legacy:EphemeralDB.
+        Defaults to False.
+
+    Returns
+    -------
+    An instance of `orion.benchmark.Benchmark`
+    """
+    setup_storage(storage=storage, debug=debug)
+
+    # fetch benchmark from db
+    db_config = _fetch_benchmark(name)
+
+    benchmark_id = None
+
+    if db_config:
+        benchmark_id, algorithms, targets = _resolve_db_config(db_config)
+
+    if not algorithms or not targets:
+        raise NoConfigurationError(
+            "Benchmark {} does not exist in DB, "
+            "algorithms and targets space was not defined.".format(name)
+        )
+
+    benchmark = _create_benchmark(name, algorithms, targets)
+
+    if benchmark_id is None:
+        # persist benchmark into db
+        _register_benchmark(benchmark)
+
+    return benchmark
+
+
+def _get_task(name, **kwargs):
+    return BenchmarkTask(of_type=name, **kwargs)
+
+
+def _get_assessment(name, **kwargs):
+    return BenchmarkAssessment(of_type=name, **kwargs)
+
+
+def _resolve_db_config(db_config):
+
+    benchmark_id = db_config["_id"]
+    algorithms = db_config["algorithms"]
+
+    obj_targets = []
+    str_targets = db_config["targets"]
+    for target in str_targets:
+        obj_target = {}
+
+        assessments = target["assess"]
+        obj_assessments = []
+        for name, parameters in assessments.items():
+            obj_assessments.append(_get_assessment(name, **parameters))
+        obj_target["assess"] = obj_assessments
+
+        tasks = target["task"]
+        obj_tasks = []
+        for name, parameters in tasks.items():
+            obj_tasks.append(_get_task(name, **parameters))
+        obj_target["task"] = obj_tasks
+
+    obj_targets.append(obj_target)
+
+    targets = obj_targets
+
+    return benchmark_id, algorithms, targets
+
+
+def _create_benchmark(name, algorithms, targets):
+
+    benchmark = Benchmark(name, algorithms, targets)
+    benchmark.setup_studies()
+
+    return benchmark
+
+
+def _create_study(benchmark, algorithms, assess, task):
+    study = Study(benchmark, algorithms, assess, task)
+    study.setup_experiments()
+
+    return study
+
+
+def _fetch_benchmark(name):
+
+    if name:
+        configs = get_storage().fetch_benchmark({"name": name})
+    else:
```

Comment:
Is there a use case for `_fetch_benchmark(None)`? I can't see any based on get_or_create_benchmark.

Reply:
I was planning `get_benchmarks ` interface for people to retrieve benchmark for just the status/view purpose, this interface will give `None` to `_fetch_benchmark` so all benchmarks will be retrieved. 

I can remove it but seems leave it won't make harm, and the `get all benchmarks` may be still necessary to added later.

Reply:
Sounds good.